src/main/java/io/r2dbc/pool/ConnectionPoolConfiguration.java

```diff
@@ -47,7 +47,7 @@
     /**
      * Constant indicating that timeout should not apply.
      */
-    public static final Duration NO_TIMEOUT = Duration.ofNanos(-1);
+    public static final Duration NO_TIMEOUT = Duration.ofMillis(-1);
```

Comment:
this change isn't strictly needed, but seems like a safer choice to avoid future similar mistakes